### PR TITLE
Feature spinner integration

### PR DESCRIPTION
**Pull request by sagar09377**
@satishshewaleglobant :

Please review the following changes:

1.added the spinner ie loader component 
2.imported spinner in cards list component
3.auto search, dropdown code in search component

### CODE REVIEW

File: Application/Client/src/app/home/components/home.component.ts [hunk 1]
```diff
@@ -23,6 +25,7 @@ export class HomeComponent implements OnInit {
   private  data : any;
   private cardUrl = 'http://in-it0289/ListingAPI/api/Listings/GetTopListings';
   private bannerUrl = 'http://in-it0289/MasterDataAPI/api/category/GetAllCategory';
+  private cardsByCategoryUrl = 'http://in-it0289/ListingAPI/api/Listings/GetListingsByCategory?Category=';
```
Comment (satishshewaleglobant) [inline Application/Client/src/app/home/components/home.component.ts:28]: REmove hard coded url and used from settings.json

File: Application/Client/src/app/filter/components/filter.component.ts [hunk 2]
```diff
@@ -58,17 +56,14 @@ export class FilterComponent {
...
-      filterData[index].isActive = true;
-    if(category === 'Top ten'){
-        this.getInitialCards.emit();
-    }else{
-     this.getCardsByCategory(category);
+  selectedOption(category, index) {
+    let filterData = this.filterData;
+    
+    for (let item of filterData) {
+      item.isActive = false;
```
Comment (satishshewaleglobant) [inline Application/Client/src/app/filter/components/filter.component.ts]: no Need for this loop , if undefined that means it is false

File: Application/Client/src/app/filter/components/filter.component.ts [hunk 3]
```diff
@@ -58,17 +56,14 @@ export class FilterComponent {
...
-      for (let item of filterData) {
-          item.isActive = false;
-      }
-      filterData[index].isActive = true;
-    if(category === 'Top ten'){
-        this.getInitialCards.emit();
-    }else{
-     this.getCardsByCategory(category);
+  selectedOption(category, index) {
+    let filterData = this.filterData;
```
Comment (satishshewaleglobant) [inline Application/Client/src/app/filter/components/filter.component.ts]: No need for extra varible use this.filterdata directlu

File: Application/Client/src/app/card-list/components/card-list.component.ts [hunk 4]
```diff
@@ -19,9 +20,16 @@ export class CardListComponent{
...
+    ngOnInit() {
         console.log('all = ',this.cards);
     }
+        
+    loading( flag ) {
+        this.isLoading = flag;
+    }
+
    showProductInfo(id){
       console.log("--------------------------------",this.cards);
```
Comment (satishshewaleglobant) [inline Application/Client/src/app/card-list/components/card-list.component.ts:34]: Remove flag

File: Application/Client/src/app/_common/loader/components/loader.component.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  selector: 'loader',
+  styles: [styles],
+  providers: [SettingsService, CService],
+  template: tpls
+})
+export class LoaderComponent {
+
+    @Input() isLoading : boolean = false;
+
+    constructor() {}
```
Comment (satishshewaleglobant) [inline Application/Client/src/app/_common/loader/components/loader.component.ts]: No Need to add empty constructor

File: Application/Client/src/app/_common/search/components/search.component.ts [hunk 6]
```diff
@@ -49,8 +50,10 @@ export class SearchComponent {
     }
 
     fetchSearchedData( text: string ) {
+        this.isLoading = true;  ;
         this._cservice.observableGetHttp( this.searchCategoryByStringUrl, null, false )
             .subscribe((res: Response) => {
+                this.isLoading = false;
```
Comment (satishshewaleglobant) [inline Application/Client/src/app/_common/search/components/search.component.ts:56]: This flag need to be false in error case also

File: Application/Client/src/app/card-list/components/card-list.component.ts [hunk 7]
```diff
@@ -19,9 +20,16 @@ export class CardListComponent{
...
 
-    ngOnInit(){
+    private isLoading:boolean = false;
+    
+    ngOnInit() {
         console.log('all = ',this.cards);
     }
+        
+    loading( flag ) {
+        this.isLoading = flag;
```
Comment (satishshewaleglobant) [inline Application/Client/src/app/card-list/components/card-list.component.ts:30]: I guess this function is not used anywhere remove this if not used

File: Application/Client/src/app/_common/loader/components/loader.component.ts (hunk 5 above)
Comment (sagar09377) [reply Application/Client/src/app/_common/loader/components/loader.component.ts]: removed the empty constructor

File: Application/Client/src/app/_common/search/components/search.component.ts (hunk 6 above)
Comment (sagar09377) [reply Application/Client/src/app/_common/search/components/search.component.ts:56]: added the flag in error case also

File: Application/Client/src/app/card-list/components/card-list.component.ts (hunk 7 above)
Comment (sagar09377) [reply Application/Client/src/app/card-list/components/card-list.component.ts:30]: it has been used , so we cannot remove it

File: Application/Client/src/app/card-list/components/card-list.component.ts (hunk 4 above)
Comment (sagar09377) [reply Application/Client/src/app/card-list/components/card-list.component.ts:34]: its needed

File: Application/Client/src/app/home/components/home.component.ts (hunk 1 above)
Comment (sagar09377) [reply Application/Client/src/app/home/components/home.component.ts:28]: setting.json no url is der

Comment (satishshewaleglobant) [reply Application/Client/src/app/home/components/home.component.ts:28]: Can u add URL to settings.json

**Pull request merged**